content/blog/end-to-end-easy-to-use-pipeline-for-training-a-model-on-medmnist-v2-using-hpe-machine-learning-development-environment-flask.md

```diff
@@ -0,0 +1,514 @@
+    # Preprocess the input data
+    processed_data = preprocess_data(image)
+
+    # Use the Determined model to make a prediction
+    output_tensor = model(processed_data)
+
+    # Convert the output tensor to a numpy array
+    output_array = output_tensor.detach().numpy()
+
+    # Apply softmax to get probabilities for each class
+    probabilities = F.softmax(output_tensor, dim=1)
+
+    # Convert the output tensor to a numpy array
+    output_array = probabilities.detach().numpy()
+
+    # Get the predicted class label
+    class_label = np.argmax(output_array)
+    class_labels = [
+        "adipose",
+        "background",
+        "debris",
+        "lymphocytes",
+        "mucus",
+        "smooth muscle",
+        "normal colon mucosa",
+        "cancer-associated stroma",
+        "colorectal adenocarcinoma epithelium",
+    ]
+
+    # Return the predicted class label as a JSON response
+    return jsonify({"prediction": class_labels[class_label]})
+```
+
+Here we define a server that accepts an image file, performs the preprocessing steps defined in the preprocess_data function, and outputs the prediction as a json object.  
+
+#### Step 2.4: Define a function to start the Flask server: 
+
+```python
+# Start the Flask server
+if __name__ == "__main__":
+    app.run(debug=True, host="0.0.0.0", port=5000)
+```
+
+In this example the server is deployed on localhost on port 5000. 
+
+#### Step 2.5: Obtain test data
+
+Your test data should be in the form of .jpeg, .png, or .jpg images, deploy the Flask server, and submit requests for inference 
+
+Full instructions for running the example can be found in the [repository](https://github.com/ighodgao/determined_medmnist_e2e). 
+
+For example, running prediction on the following image of adipose tissue results in: 
+
+![](/img/adipose.jpeg)
+
+`"prediction": "adipose"`
+
+Deploying your model can make it a lot easier to use in production, and while this example uses Flask as a simple proof-of-concept, tools like KServe can help deploy your models in production. 
+
+### Conclusion  
+
+That’s it! We have shown how to implement an end-to-end machine learning workflow using HPE Machine Learning Development Environment and Flask. For more information about HPE’s AI toolkit please visit the [HPE AI Solutions homepage here.](https://www.hpe.com/us/en/solutions/artificial-intelligence.html)  
+
+If you enjoyed reviewing this model training and deployment example, [we invite you to get in touch with the Determined Community](https://join.slack.com/t/determined-community/shared_invite/zt-1txc10qgy-yJ2puE6DxgrhdH9TIK93tw) and try this out for yourself by following the Determined [Documentation](https://docs.determined.ai/latest/).  
+
+If you and your team are ready for premium machine learning support from HPE, please contact us at the [HPE Machine Learning Development Environment homepage](https://www.hpe.com/us/en/solutions/artificial-intelligence/machine-learning-development-environment.html).
```

Comment:
I recommend changing "please contact us at the" to "please contact the team via the"